### PR TITLE
Adds vertical regridding feature

### PR DESCRIPTION
**Pull request by navidcy**
This PR adds a vertical regrid feature + tests.

cc @adelinehillier

### CODE REVIEW

**Comment by navidcy** [comment]
@glwagner, what was those lines in 846cd69 all about? Put them back in if they are crucial...

File: src/Fields/regridding_fields.jl [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+     x           x
+                --- ks = 4
+    --- kt=3     x 
+                --- ks = 3
+     x           x k=2
+                --- ks = 2
+    --- kt=2     x k=1
+                --- ks = 1
+
+"""
```
Comment (navidcy) [inline src/Fields/regridding_fields.jl:45]: @glwagner delete these?

File: src/Fields/regridding_fields.jl [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    end
+
+    if Lz(c) == Lz(d) && Nz(c) == Nz(d)
+        return _set_similar_fields!(c, d)
+    else
+        return interp_and_set!(c, d)
+    end
+
+end
+=#
```
Comment (navidcy) [inline src/Fields/regridding_fields.jl:174]:
```suggestion
```
?

**Comment by navidcy** [comment]
1db7a62 -> I think #1986 deserves its own PR. There are a lot of changes in the doctests that need to come with this change.

**Comment by navidcy** [comment]
@glwagner I think there are some issues on the GPU; could you have a look?

File: src/Fields/regridding_fields.jl (hunk 2 above)
Comment (adelinehillier) [reply src/Fields/regridding_fields.jl:174]: This was the old regridding code for 1-D--can delete!

File: src/Fields/regridding_fields.jl [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+     x           x
+                --- ks = 4
+    --- kt=3     x 
+                --- ks = 3
+     x           x k=2
+                --- ks = 2
+    --- kt=2     x k=1
+                --- ks = 1
+
+"""
```
Comment (navidcy) [inline src/Fields/regridding_fields.jl:45]:
```suggestion
```

File: test/test_field.jl [hunk 4]
```diff
@@ -289,4 +289,78 @@ end
...
+            super_fine_column_regular_grid   = RegularRectilinearGrid(size=5, z=(0, Lz), topology=topology)
+            
+            coarse_column_regular_c       = CenterField(arch, coarse_column_regular_grid)
+            fine_column_regular_c         = CenterField(arch, fine_column_regular_grid)
+            fine_column_stretched_c       = CenterField(arch, fine_column_stretched_grid)
+            very_fine_column_stretched_c  = CenterField(arch, very_fine_column_stretched_grid)
+            super_fine_column_stretched_c = CenterField(arch, super_fine_column_stretched_grid)
+            super_fine_column_regular_c   = CenterField(arch, super_fine_column_regular_grid)
+
+            # we initialize an array on the `fine_column_stretched_grid` and regrid it to the rest
```
Comment (navidcy) [inline test/test_field.jl:315]:
```suggestion
            # we initialize an array on the `fine_column_stretched_grid`, regrid it to the rest
            # grids, and check whether we get the anticipated results

```

File: src/Fields/regridding_fields.jl (hunk 2 above)
Comment (navidcy) [reply src/Fields/regridding_fields.jl]: Thanks :)

File: Project.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 name = "Oceananigans"
 uuid = "9e8cae18-63c1-5223-a75c-80ca9d6e9a09"
-version = "0.62.2"
+version = "0.62.3"
```
Comment (tomchor) [inline Project.toml:3]: This is a new feature. Shouldn't we bump the version to 0.63.0 instead? Or are there bug fixes?

**Comment by navidcy** [comment]
That worked @glwagner. 

Any approvals by anyone?

**Comment by tomchor** [comment]
I didn't see any docstring explaining what's actually being done. Is it possible to add some? Maybe with a minimal example? (Unless I missed it, in which case please ignore the comment.)

Also, if I understand correctly, this only regrids in the `z` direction, no? Can we change the same from simply `regrid` to something more specific? Like `regrid_z` or `vertical_regrid`?

**Comment by glwagner** [comment]
> Also, if I understand correctly, this only regrids in the `z` direction, no? Can we change the same from simply `regrid` to something more specific? Like `regrid_z` or `vertical_regrid`?

The idea is that we will eventually support general regridding. In this PR, we only implement vertical regridding as a start.

**Comment by tomchor** [comment]
> > Also, if I understand correctly, this only regrids in the `z` direction, no? Can we change the same from simply `regrid` to something more specific? Like `regrid_z` or `vertical_regrid`?
> 
> The idea is that we will eventually support general regridding. In this PR, we only implement vertical regridding as a start. The next case to support would be regridding for Cartesian grids; after that, for curvilinear grids. We've attempted to write an informative error:
> 
> ```julia
>     msg = """Regridding
>              $(short_show(v)) on $(short_show(source_grid))
>              to $(short_show(u)) on $(short_show(target_grid))
>              is not supported."""
> ```

I see! I got a bit confused about this error because I'm not familiar with `SingleColumnGrid`, but it makes more sense now.

**Review by tomchor — APPROVED** [review]
This looks good! I'll approve it now but I'll leave the following suggestions if you think they're helpful:

- Add brief docstrings with a quick example
- Change the names from `u` and `v` in the functions to something more general in order to avoid confusion. Maybe `a`, `b` or some greek letters. (I'm assuming that this works for any `Field`s, no?)

**Comment by navidcy** [comment]
> This looks good! I'll approve it now but I'll leave the following suggestions if you think they're helpful:
> 
> * Add brief docstrings with a quick example
> * Change the names from `u` and `v` in the functions to something more general in order to avoid confusion. Maybe `a`, `b` or some greek letters. (I'm assuming that this works for any `Field`s, no?)

I agree. Sorry that seems "rushed" but we need this feature for some project. But I will put these as an issue!

**Comment by tomchor** [comment]
> > This looks good! I'll approve it now but I'll leave the following suggestions if you think they're helpful:
> > 
> > * Add brief docstrings with a quick example
> > * Change the names from `u` and `v` in the functions to something more general in order to avoid confusion. Maybe `a`, `b` or some greek letters. (I'm assuming that this works for any `Field`s, no?)
> 
> I agree. Sorry that seems "rushed" but we need this feature for some project. But I will put these as an issue!

That's alright. Thanks for creating https://github.com/CliMA/Oceananigans.jl/issues/1991

**Pull request merged**